website/www/site/content/en/documentation/basics.md

```diff
@@ -365,6 +369,108 @@ For more information about runners, see the following pages:
  * [Beam Capability Matrix](/documentation/runners/capability-matrix/)
 
+### Trigger
+
+When collecting and grouping data into windows, Beam uses _triggers_ to
+determine when to emit the aggregated results of each window (referred to as a
+_pane_). If you use Beam’s default windowing configuration and default trigger,
+Beam outputs the aggregated result when it estimates all data has arrived, and
+discards all subsequent data for that window.
+
+At a high level, triggers provide two additional capabilities compared to
+outputting at the end of a window:
+
+ 1. Triggers allow Beam to emit early results, before all the data in a given
+    window has arrived. For example, emitting after a certain amount of time
+    elapses, or after a certain number of elements arrives.
+ 2. Triggers allow processing of late data by triggering after the event time
+    watermark passes the end of the window.
+
+These capabilities allow you to control the flow of your data and also balance
+between data completeness, latency, and cost.
+
+Beam provides a number of pre-built triggers that you can set:
+
+ * **Event time triggers**: These triggers operate on the event time, as
+   indicated by the timestamp on each data element. Beam’s default trigger is
+   event time-based.
+ * **Processing time triggers**: These triggers operate on the processing time,
+   which is the time when the data element is processed at any given stage in
+   the pipeline.
+ * **Data-driven triggers**: These triggers operate by examining the data as it
+   arrives in each window, and firing when that data meets a certain property.
+   Currently, data-driven triggers only support firing after a certain number of
+   data elements.
+ * **Composite triggers**: These triggers combine multiple triggers in various
+   ways.
+
+For more information about triggers, see the following page:
+
+ * [Beam Programming Guide: Triggers](/documentation/programming-guide/#triggers)
+
+### State and timers
+
+Beam’s windowing and triggers provide an abstraction for grouping and
+aggregating unbounded input data based on timestamps. However, there are
+aggregation use cases that might require an even higher degree of control. State
+and timers are two important concepts that help with these uses cases.
+
+**State**:
+
+Beam provides the State API for manually managing per-key state, allowing for
+fine-grained control over aggregations.  The State API lets you augment
+element-wise operations (for example, `ParDo` or `Map`) with mutable state.
+
+The State API models state per key. To use the state API, you start out with a
+keyed `PCollection`. A `ParDo` that processes this `PCollection` can declare
+persistent state variables. When you process each element inside the `ParDo`,
+you can use the state variables to write or update state for the current key or
+to read previous state written for that key. State is always fully scoped only
+to the current processing key.
+
+Beam provides several types of state:
```

Comment:
Now we've got MapState and SetState and even OrderedListState. (of course documenting some is better than none!)

Reply:
I added super brief mention of those and reorganized this a bit. I grouped bag/map/set/orderedlist with "A common use case for state is to accumulate multiple elements into a group:"  I used "group" because I was worried about using "collection" and causing confusion  (PCollection vs. collection), what do you think?

Reply:
+1 to mention of other states, we should also mention that not all states maybe supported by all runners.